main/src/main/java/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1847,6 +1857,114 @@ public void setContent() {
 
         }
 
+        private void identifyTranslationLanguage() {
```

Comment:
Please move the translation and related logic (eg download of language packages including related dialogs) to its own new class (maybe a new Utils class). That way we can easily reuse this logic in other places in the future eg waypoint texts, logs, trackable listings and whatever will come to our mind in the future. The utility class would ideally offer only very simple methods like "getTranslateLanguage" (null if none), "translate" (with parameters of type String and Context; the later in case dialogs are needed).

At some point (not necessarily with first PR) we need to allow users to manage/delete downloaded translation packages. Logic to access/list and delete those can then also be encapsulated in said Utils class

Reply:
See the todo-list in PR description 😉